.github/workflows/justtest.yaml

```diff
@@ -0,0 +1,699 @@
+      #
+      # actions/checkout has no option to select submodules to
+      # fetch, so perform clone without submodules and fetch them
+      # in the next step.
+      #
+      # We don't need the commit history, so a shallow clone is
+      # OK.
+      - name: Checkout repository (shallow clone, no submodules)
+        uses: actions/checkout@v4
+        with:
+          # See actions/checkout step in the build job above.
+          persist-credentials: false
+
+      # test-run is needed, because it is the test runner we're
+      # using.
+      #
+      # checks and metrics are needed, because we run tests from
+      # these submodules as part of tarantool's test suite.
+      - name: Checkout test-run, checks and metrics submodules
+        run: |
+          git submodule update --init --recursive \
+            test-run                              \
+            third_party/checks                    \
+            third_party/metrics
+
+      - name: Download build artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: build-${{ matrix.build-conf }}
+
+      - name: Restore executable bits
+        run: chmod a+x src/tarantool src/lib/small/test/*.test
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: pip
+          cache-dependency-path: test-run/requirements.txt
+
+      - name: Setup testing dependencies
+        run: pip install -r test-run/requirements.txt
+
+      # --retry
+      # -------
+      #
+      # Zero tolerance about unstable tests would be nice, but it
+      # doesn't work good for us, at least for now.
+      #
+      # A developer who mets the flaky failure is not necessarily
+      # one who is responsible for the given subsystem. So, he/she
+      # reports the unstable test and restarts the job. It takes
+      # some human attention and time, so we're interested in
+      # minimizing such situations.
+      #
+      # Even if an author of the test mets the unstable failure,
+      # he/she may have other ongoing work now. It is often not OK
+      # to stop everything, lost needed context from the head and
+      # work on fixing the test or a problem in tarantool that it
+      # reveals.
+      #
+      # In short, a flaky problem in one subsystem shouldn't block
+      # the development of other subsystems.
+      #
+      # This way we add some tolerance for unstable tests using
+      # the --retry option.
+      #
+      # --long
+      # ------
+      #
+      # Run all the tests, including ones that are marked as long.
+      #
+      # TODO: Drop --long when gh-11043 lands.
+      - name: Run ${{ matrix.test_suite }}/*
+        run: ./test/test-run.py --retries 3 ${{ matrix.test_suite }}/ --long
+
+  # }}} Run tarantool testing (in separate jobs)
+
+  # {{{ Run LuaJIT testing (in separate jobs)
+
+  luajit-test:
+    runs-on: ubuntu-latest
+    needs: build
+
+    strategy:
```

Comment:
I suppose it would be necessary to add `fail-fast: false` to all tests to avoid rerunning all suites if at least one failed.
Maybe the same thing should be done for the build matrix, but I'm not sure.